actions/test/run_local.ts

```diff
@@ -72,58 +87,124 @@ async function processBlock(
   );
   let hasErrors = false;
   for (const transaction of blockWithTransactions.transactions) {
-    const receipt = await provider.getTransactionReceipt(transaction.hash);
-    if (receipt) {
-      const {
-        hash,
-        from,
-        value,
-        nonce,
-        gasLimit,
-        maxPriorityFeePerGas,
-        maxFeePerGas,
-      } = transaction;
-
-      const testTransactionEvent: TestTransactionEvent = {
-        blockHash: block.hash,
-        blockNumber: block.number,
-        from,
-        hash,
-        network: chainId.toString(),
-        logs: receipt.logs,
-        input: "",
-        value: value.toString(),
-        nonce: nonce.toString(),
-        gas: gasLimit.toString(),
-        gasUsed: receipt.gasUsed.toString(),
-        cumulativeGasUsed: receipt.cumulativeGasUsed.toString(),
-        gasPrice: receipt.effectiveGasPrice.toString(),
-        gasTipCap: maxPriorityFeePerGas ? maxPriorityFeePerGas.toString() : "",
-        gasFeeCap: maxFeePerGas ? maxFeePerGas.toString() : "",
-        transactionHash: transaction.hash,
-      };
-
-      // run action
-      console.log(`[run_local] Run "addContract" action for TX ${hash}`);
-      const result = await testRuntime
-        .execute(addContract, testTransactionEvent)
-        .then(() => true)
-        .catch((e) => {
-          hasErrors = true;
-          console.error(
-            `[run_local] Error running "addContract" action for TX:`,
-            e
-          );
-          return false;
-        });
-      console.log(
-        `[run_local] Result of "addContract" action for TX ${hash}: ${_formatResult(
-          result
-        )}`
-      );
-    }
+    hasErrors ||= !(await !_processTx(
+      provider,
+      block,
+      chainId,
+      testRuntime,
+      transaction
+    ));
+  }
+
+  hasErrors ||= !(await _pollAndPost({ block, chainId, testRuntime }));
+
+  if (hasErrors) {
+    throw new Error("[run_local] Errors found in processing block");
```

Comment:
Does it make sense to also log the block number where this failed for extra clarity?

```suggestion
    throw new Error(`[run_local] Errors found in processing block: ${block}`);
```

Reply:
It does!